pkg/imgpkg/registry/auth/env_keychain.go

```diff
@@ -54,7 +55,7 @@ func (k *EnvKeychain) Resolve(target regauthn.Resource) (regauthn.Authenticator,
 	}
 
 	for _, info := range infos {
-		registryURLMatches, err := credentialprovider.URLsMatchStr(info.URL, target.String())
+		registryURLMatches, err := urlsMatchStr(info.URL, target.String())
```

Comment:
This seems a little unfortunate but is what GGCR did as well https://github.com/google/go-containerregistry/blob/2b1087abc076a6e82339480ea191b7245962932c/pkg/authn/kubernetes/keychain.go#L187

Reply:
Oh yeah, I omitted the full context there accidentally. Kubernetes [has the original version of this code](https://github.com/kubernetes/kubernetes/blob/b2ed6ca64f55452cdb4063d62920dddccd4344f7/pkg/credentialprovider/keyring.go#L184-L194), it was extracted out into vdemeester/k8s-pkg-credentialprovider awaiting https://github.com/kubernetes/enhancements/pull/1406, I'm not exactly sure where that ended up but now GGCR has copied that code into their codebase for their Kubernetes keychain implementation

Reply:
We might be able to simplify this code, I just did the lazy option of copying all these interconnected functions